### PR TITLE
Large Vector tests for DGL Ops Part 2

### PR DESCRIPTION
**Pull request by ChaiBapchya**
## Description ##
add hyperbolic, logical, sign and regression tests for large vector

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- [x] Code is well-documented: 
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Add tests to test_large_array.py
- [x] Fix issue with test_concat introduced in PR #15960

### CODE REVIEW

**Comment by ChaiBapchya** [comment]
```
test_large_array.test_gluon_embedding ... ok
test_large_array.test_ndarray_zeros ... ok
test_large_array.test_ndarray_ones ... ok
test_large_array.test_ndarray_convert ... ok
test_large_array.test_ndarray_random_uniform ... ok
test_large_array.test_ndarray_random_randint ... ok
test_large_array.test_ndarray_random_exponential ... ok
test_large_array.test_ndarray_random_gamma ... ok
test_large_array.test_ndarray_random_multinomial ... ok
test_large_array.test_ndarray_random_generalized_negative_binomial ... ok
test_large_array.test_ndarray_random_negative_binomial ... ok
test_large_array.test_ndarray_random_normal ... ok
test_large_array.test_ndarray_random_poisson ... ok
test_large_array.test_ndarray_random_randn ... ok
test_large_array.test_ndarray_random_shuffle ... ok
test_large_array.test_ndarray_empty ... ok
test_large_array.test_elementwise ... ok
test_large_array.test_reduce ... ok
test_large_array.test_dot ... ok
test_large_array.test_FullyConnected ... ok
test_large_array.test_broadcast ... ok
test_large_array.test_clip ... ok
test_large_array.test_split ... ok
test_large_array.test_argmin ... ok
test_large_array.test_tile ... ok
test_large_array.test_take ... ok
test_large_array.test_slice ... ok
test_large_array.test_slice_assign ... ok
test_large_array.test_expand_dims ... ok
Helper function that cleans up memory by releasing it from memory pool ... ok
test_large_array.test_squeeze ... ok
test_large_array.test_broadcast_div ... ok
test_large_array.test_Dense ... ok
test_large_array.test_where ... ok
test_large_array.test_pick ... ok
test_large_array.test_depthtospace ... ok
test_large_array.test_spacetodepth ... ok
test_large_array.test_diag ... ok
test_large_array.test_ravel_multi_index ... ok
test_large_array.test_unravel_index ... ok
test_large_array.test_transpose ... ok
test_large_array.test_swapaxes ... ok
test_large_array.test_flip ... ok
test_large_array.test_softmax ... ok
test_large_array.test_argsort ... ok
test_large_array.test_sort ... ok
test_large_array.test_topk ... ok
test_large_array.test_exponent_logarithm_operators ... ok
test_large_array.test_power_operators ... ok
test_large_array.test_sequence_mask ... ok
test_large_array.test_sequence_reverse ... ok
test_large_array.test_sequence_last ... ok
test_large_array.test_softmax_cross_entropy ... ok
test_large_array.test_index_copy ... ok
test_large_array.testSoftmaxOutput ... ok
test_large_array.test_leaky_relu ... ok
test_large_array.test_pooling ... ok
test_large_array.test_layer_norm ... ok
test_large_array.test_dropout ...
```

**Comment by ChaiBapchya** [comment]
@access2rohit @anirudh2290 @zheng-da PR ready for review

File: tests/nightly/test_large_vector.py [hunk 1]
```diff
@@ -708,6 +708,111 @@ def test_full():
     assert a[-1] == 3
 
 
+def test_hyperbolic():
+    def test_arccosh(a):
```
Comment (anirudh2290) [inline tests/nightly/test_large_vector.py]: can this be merged with trignometric functions that are tested in this PR  ? https://github.com/apache/incubator-mxnet/pull/16477

File: tests/nightly/test_large_vector.py [hunk 2]
```diff
@@ -708,6 +708,111 @@ def test_full():
...
+    assert_almost_equal(mx_res[-1].asnumpy(), np.sign(a[-1].asnumpy()))
+
+
+def test_logical():
+    def test_logical_and(a, b):
+        mx_res = mx.nd.logical_and(a, b)
+        assert_almost_equal(mx_res[-1].asnumpy(), np.logical_and(a[-1].asnumpy(), b[-1].asnumpy()))
+
+    def test_logical_or(a, b):
+        mx_res = mx.nd.logical_and(a, b)
```
Comment (anirudh2290) [inline tests/nightly/test_large_vector.py]: all testing logical_and ?

File: tests/nightly/test_large_vector.py [hunk 3]
```diff
@@ -708,6 +708,111 @@ def test_full():
...
+        assert_almost_equal(mx_res[-1].asnumpy(), np.logical_and(a[-1].asnumpy(), b[-1].asnumpy()))
+
+    def test_logical_not(a, b):
+        mx_res = mx.nd.logical_and(a, b)
+        assert_almost_equal(mx_res[-1].asnumpy(), np.logical_and(a[-1].asnumpy(), b[-1].asnumpy()))
+
+    def test_logical_xor(a, b):
```
Comment (anirudh2290) [inline tests/nightly/test_large_vector.py]: prefer check_logical_xor instead of test_logical_xor

File: tests/nightly/test_large_vector.py (hunk 1 above)
Comment (access2rohit) [reply tests/nightly/test_large_vector.py]: can we merge those later ? We divided the work like this

Comment (anirudh2290) [reply tests/nightly/test_large_vector.py]: this should be quick right ?

**Comment by anirudh2290** [comment]
Can you also address : https://github.com/apache/incubator-mxnet/pull/16104#discussion_r333275164

File: tests/nightly/test_large_vector.py (hunk 2 above)
Comment (ChaiBapchya) [reply tests/nightly/test_large_vector.py]: good catch!

File: tests/nightly/test_large_vector.py (hunk 3 above)
Comment (ChaiBapchya) [reply tests/nightly/test_large_vector.py]: cool

File: tests/nightly/test_large_vector.py (hunk 1 above)
Comment (ChaiBapchya) [reply tests/nightly/test_large_vector.py]: Done!

**Comment by anirudh2290** [comment]
Can you address this comment too : https://github.com/apache/incubator-mxnet/pull/16497#issuecomment-542953179

**Comment by ChaiBapchya** [comment]
Tried to convey the message offline. Doing it again here. 
In the PR #16229 2 tests related to transpose had been introduced. This could have been "merged" inside the original transpose but it was kept separate. If that's the case, I find it inconsistent to have this one merged. Do you suggest including all "transpose" (including test_big_ related functions under one "test_pseudo2dtranspose", "test_big_transpose"

**Comment by access2rohit** [comment]
@anirudh2290 can we merge this PR now ?

**Comment by anirudh2290** [comment]
I think its okay to keep it as it is. Just add the with_seed back in test_large_transpose.

File: tests/nightly/test_large_vector.py [hunk 4]
```diff
@@ -708,6 +708,70 @@ def test_full():
...
+        mx_res = mx.nd.logical_or(a, b)
+        assert_almost_equal(mx_res[-1].asnumpy(), np.logical_and(a[-1].asnumpy(), b[-1].asnumpy()))
+
+    def check_logical_not(a, b):
+        mx_res = mx.nd.logical_not(a, b)
+        assert_almost_equal(mx_res[-1].asnumpy(), np.logical_and(a[-1].asnumpy(), b[-1].asnumpy()))
+
+    def check_logical_xor(a, b):
+        mx_res = mx.nd.logical_xor(a, b)
+        assert_almost_equal(mx_res[-1].asnumpy(), np.logical_and(a[-1].asnumpy(), b[-1].asnumpy()))
```
Comment (anirudh2290) [inline tests/nightly/test_large_vector.py]: checks with logical_and for all tests ?

**Comment by anirudh2290** [comment]
are the symbolic tests that you mentioned offline going to be added in a different PR ?

**Comment by ChaiBapchya** [comment]
Ya. So we don't have symbolic specific tests. test_regression introduced in this PR uses symbolic with individual dim size > 2**32
There are couple of other places (test_large_array.layer_norm) which uses symbolic (but it didn't fail coz individual dim sie was still under 2**32)

**Pull request merged**